### ISSUE TITLE
Linux kernel 6.5 compile failure due to changes with ioprio

### ISSUE THREAD

**Issue by garybuhrmaster** [MythTV/mythtv#775]
Just a heads up:

The upcoming (~late August 2023) Linux 6.5 kernel has merged changes to the ioprio macros such that a compilation now fails with:

```
In file included from mythmiscutil.cpp:705:
/usr/include/linux/ioprio.h:110:43: error: expected primary-expression before 'int'
  110 | static __always_inline __u16 ioprio_value(int class, int level, int hint)
      |                                           ^~~
/usr/include/linux/ioprio.h:110:54: error: expected primary-expression before 'int'
  110 | static __always_inline __u16 ioprio_value(int class, int level, int hint)
      |                                                      ^~~
/usr/include/linux/ioprio.h:110:65: error: expected primary-expression before 'int'
  110 | static __always_inline __u16 ioprio_value(int class, int level, int hint)
      |                                                                 ^~~
/usr/include/linux/ioprio.h:110:73: error: expression list treated as compound expression in initializer [-fpermissive]
  110 | static __always_inline __u16 ioprio_value(int class, int level, int hint)
      |                                                                         ^
mythmiscutil.cpp: In function 'bool myth_ioprio(int)':
mythmiscutil.cpp:726:22: error: 'ioprio_value' cannot be used as a function
  726 |     int new_ioprio = IOPRIO_PRIO_VALUE(new_ioclass, new_iodata);
      |                      ^~~~~~~~~~~~~~~~~
mythmiscutil.cpp:738:22: error: 'ioprio_value' cannot be used as a function
  738 |         new_ioprio = IOPRIO_PRIO_VALUE(IOPRIO_CLASS_BE, new_iodata);
      |                      ^~~~~~~~~~~~~~~~~
make[2]: *** [Makefile:1743: obj/mythmiscutil.o] Error 1
```

I have not looked further at either the kernel changes, nor the usage in mythmiscutil.cpp (i.e. this is just an initial heads up).

Once the projects BuildBot Fedora Rawhide systems get updated to include the 6.5rc kernel I expect the project will start to see build failures.

**Comment by linuxdude42** [comment]
Looks like maybe its just missing the definition of "int"?

**Comment by linuxdude42** [comment]
This isn't a problem caused by mythtv.  The following trivial program also fails to compile:

```
#include <linux/ioprio.h>

int main (int argc, char **argv)
{
}
```
The first four errors are the same for both compiles. The last two mythtv errors are a cascading problem caused by the first four errors.

**Comment by linuxdude42** [comment]
This header uses the C++ reserved keyword "class".

**Comment by garybuhrmaster** [comment]
> This header uses the C++ reserved keyword "class".

Good detective work.  I would not be surprised if that breaks other apps at some point in the future that utilize the ioprio headers.

I suppose if one does not want to split out that usage to a pure c compilation unit one can do something like (untested):
```
#define class ioprioclass
#include <linux/ioprio.h>
#undef class
```
but that feels somewhat fragile.

**Comment by linuxdude42** [comment]
I just committed something like that a few minutes ago. There are a couple of other places in the code that have similar #define/#undef wrappers where declarations in X headers conflict with Qt headers.

**Comment by bennettpeter** [comment]
What about ? 
extern "C" {
  #include <linux/ioprio.h>
}
This would be less of a hack, if it works.

**Comment by linuxdude42** [comment]
Unfortunately, it doesn't work.

**Comment by linuxdude42** [comment]
Fixed in 7d6c7d605f.

**Comment by mspieth** [comment]
What I used in android is to place a file in the same path in the source tree with a -I or -isystem (or create a "linux" dir in one of the existing ones) and then use #include_next. The file would be

```
#define class class2
#include_next <linux/iopriv.h>
#undef class
```
This makes it completely transparent and easy to unwind.
This only works for clang and gcc of course since its a gcc extension.
